src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java

```diff
@@ -120,6 +121,21 @@ protected boolean isElementOnPage(WebElement element) {
   }
 
   /**
+   * Method to check if WebElement is displayed on the page
+   * @param element
+   * @return true if element is displayed, otherwise return false
+   */
+
+  protected boolean isElementDisplayed(WebElement element) {
+    try {
+      return element.isDisplayed();
+    } catch (NoSuchElementException e) {

```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s1/coding_rules#rule_key=squid%3AS1166)